packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js

```diff
@@ -180,26 +180,28 @@ if (__DEV__) {
       return true;
     }
 
-    if (typeof value === 'boolean' && value === false) {
-      warning(
-        DOMProperty.shouldAttributeAcceptBooleanValue(name),
-        'If you mean to conditionally pass an attribute, use a ternary ' +
-          'expression: `%s`={condition ? value : undefined} instead of ' +
-          '{condition && value}.%s',
-        value,
-        getStackAddendum(),
-      );
+    if (typeof value === 'boolean') {
+      if (value === true) {
+        warning(
+          DOMProperty.shouldAttributeAcceptBooleanValue(name),
```

Comment:
Can we make this `warning(false` and move out this (negated) condition together with the top one?

Reply:
Yes :) Just so I have a correct understanding, the refactor will look like:
```
if (typeof value === 'boolean' && !false) {
  if (value === true) {
    warning(false,
      ...
    );
  } else {
    warning(false,
      ...
    );
  }
  ...
}
```
Is this the branching logic you prefer?

Reply:
```js
if (typeof value === 'boolean' && !DOMProperty.shouldAttributeAcceptBooleanValue(name)) {
  if (value === true) {
    warning(false,
      ...
    );
  } else {
    warning(false,
      ...
    );
  }
  ...
}